src/trainers/KNNTrainer.js

```diff
@@ -29,34 +29,51 @@ export default class KNNTrainer {
   startTraining(store) {
     this.store = store;
     const state = store.getState();
+    const datasetSize = state.data.length;
+    const defaultRegressionK = datasetSize < 100 ? 1 : 5;
```

Comment:
It would be good to have some summary commentary capturing why `100`, `5`,  and `11` below, which all appear to be chosen deliberately...

Reply:
Good idea. No magic numbers! Updated ✔️